File: Euler 1-10/task10.c
#include <stdio.h>
#include <stdlib.h>


int isPrime(long int * array, long int num, long int len);
long int findSum(long int * array, long int len);


int main()
{
    long int count = 0;
    long int * primes = (long int *)malloc(2000000*sizeof(long int));
    for (long int i = 2; i < 2000000; i++)    
    {
        if (isPrime(primes, i, count))
        {
            primes[count] = i;
            count++;
        }
    }
    printf("%ld\n", findSum(primes, count));
    return 0;
}


int isPrime(long int * array, long int num, long int len)
{
    for (int i = 0; i < len; i++)
        if (num % array[i] == 0)
            return 0;
    return 1;
}


long int findSum(long int * array, long int len)
{
    long int sum = 0;
    for (long int i = 0; i < len; i++)
        sum += array[i];
    return sum;
}

File: Euler 1-10/task10.py
from math import sqrt
from time import time


def is_prime(num, lst):
    for i in lst:
        if i > sqrt(num):
            break
        if num % i == 0:
            return False
    return True


def fill_primes(num, lst):
    for i in range(lst[-1] + 1, num):
        if is_prime(i, lst):
            lst.append(i)
    return lst


def find_sum(limit):
    primes = fill_primes(limit, [2, 3])
    return sum(primes)


start = time()
print(find_sum(2000000))
print("Time {} s.".format(time() - start))